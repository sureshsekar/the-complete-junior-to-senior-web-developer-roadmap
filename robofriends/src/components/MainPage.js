import React, { Component } from "react";
import Header from "../components/Header";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import "./MainPage.css";
import ErrorBoundry from "../components/ErrorBoundry";




class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

filterRobots = ( ) => {
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    });}
  render() {
    const {  onSearchChange, robots, isPending } = this.props;

    return !isPending ? (
      <h1>Loading</h1>
    ) : (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundry>
          <CardList robots={this.filterRobots()} />
          </ErrorBoundry>
        </Scroll>
      </div>
    );
  }
}

export default MainPage
