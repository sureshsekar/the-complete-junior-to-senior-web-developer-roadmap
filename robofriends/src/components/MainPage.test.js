import { shallow, mount, render } from "enzyme";
import React from "react";
import MainPage from "./MainPage";


let wrapper
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false

	}
	wrapper = shallow(<MainPage { ...mockProps } />)
})

it('renders MainPage withou crashing', () => {
	expect(wrapper).toMatchSnapshot();
})

it('filters robots correctly', () => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id: 3,
			name: 'John',
			email: 'john@gmail.com'
		}],
		searchField: 'john',
		isPending: false

	}
	const wrapper2 = shallow(<MainPage { ...mockProps2 } />)
	expect(wrapper2.instance().filterRobots()).toEqual([{
			id: 3,
			name: 'John',
			email: 'john@gmail.com'
		}])
})