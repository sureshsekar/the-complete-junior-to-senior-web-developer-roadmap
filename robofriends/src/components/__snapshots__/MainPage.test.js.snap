// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage withou crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobots={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    robots={Array []}
    searchField=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading",
    },
    "ref": null,
    "rendered": "Loading",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading",
      },
      "ref": null,
      "rendered": "Loading",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
